ocs_ci/ocs/resources/mcg.py

```diff
@@ -799,20 +799,41 @@ def _check_mirroring():
                     else:
                         results.append(False)
 
-            return all(results)
-
-        try:
-            for mirroring_is_complete in TimeoutSampler(timeout, 5, _check_mirroring):
-                if mirroring_is_complete:
+            return results
+
+        is_mirroring_complete = _check_mirroring()
+        current_percentage = (
+            is_mirroring_complete.count(True) / len(is_mirroring_complete)
+        ) * 100
```

Comment:
Also, both the names `_is_mirroring_complete` and `_check_mirroring()` are VERY misleading here:

- `is_mirroring_complete`: any variable name that starts with "is" is expected to be a boolean, but this is a list! A more fitting name for the existing implementation would be "mirror_status_flags" or even "mirror_complete_bool_list" to make it really obvious.

- `_check_mirroring`: this name was okay-ish for the original  implementation where it returned a bool, but now that it returns a list the name should give some sort of hint for this returned value. Something like `_get_mirroring_status` would suit better here.

Reply:
Finally, since `_check_mirroring()` is only used in this the scope of `check_if_mirroring_is_done()`, you can and should just modify it to return the result of this repeating logic, and change it's name to "_get_mirroring_percentage". Then you can call `_get_mirroring_percentage()` instead of repeating these 4 lines.